adblock/rules_test.go

```diff
@@ -138,13 +138,19 @@ func TestDomainAnchor(t *testing.T) {
 func TestOptsDomain(t *testing.T) {
 	testInputs(t, `
 /ads$domain=foo.com|~info.foo.com
+||bar.com^$domain=bar.com
+||bar.com^$domain=baz.com
 `,
 		[]TestInput{
-			{URL: "http://foo.com/ads", Matched: true},
-			{URL: "http://other.foo.com/ads", Matched: true},
-			{URL: "http://info.foo.com/ads", Matched: false},
-			{URL: "http://foo.com/img", Matched: false},
+			{URL: "http://foo.com/ads", Matched: true, OriginDomain: "foo.com"},
```

Comment:
If I were writing the tests from scratch yes, at this point I prefer to preserve the existing coverage, I believe the different domain case is handled by the new tests below.